### PR TITLE
BSP-2228: Adds the ability to sort in multiple orderings (ascending/descending).

### PR DESCRIPTION
**Pull request by rohitmonga**

### CODE REVIEW

**Comment by rohitmonga** [comment]
Updated with changes.

File: db/src/main/java/com/psddev/cms/tool/Search.java [hunk 1]
```diff
@@ -208,6 +214,25 @@ public Search(ToolPageContext page, Iterable<UUID> typeIds) {
         setNewItemIds(new LinkedHashSet<>(page.params(UUID.class, NEW_ITEM_IDS_PARAMETER)));
         setIgnoreSite(page.param(boolean.class, IGNORE_SITE_PARAMETER));
 
+        String sort = page.param(String.class, SORT_PARAMETER);
```
Comment (danslaughter2) [inline db/src/main/java/com/psddev/cms/tool/Search.java]: This logic should probably be encapsulated in the `#setSort(String)` method.

File: db/src/main/java/com/psddev/cms/tool/Search.java [hunk 2]
```diff
@@ -474,8 +520,13 @@ public String createFrameName(String prefix) {
...
         }
 
-        addSorts(sorts, selectedType);
-        addSorts(sorts, Database.Static.getDefault().getEnvironment());
+        try {
+            addSorts(sorts, selectedType);
+            addSorts(sorts, Database.Static.getDefault().getEnvironment());
+
+        } catch (IOException error) {
+            // Ignore.
```
Comment (danslaughter2) [inline db/src/main/java/com/psddev/cms/tool/Search.java]: Exceptions generally shouldn't be ignored/suppressed. And in this case should probably be handled inside the `#addSorts(Map, ObjectStruct)` API.

File: db/src/main/java/com/psddev/cms/tool/search/AbstractSearchResultView.java [hunk 3]
```diff
@@ -105,32 +105,32 @@ protected ObjectField updateSort() {
         ObjectType selectedType = search.getSelectedType();
         ToolUi ui = selectedType == null ? null : selectedType.as(ToolUi.class);
 
-        if (search.getSort() == null) {
+        if (search.getEffectiveSort() == null) {
             if (ui != null && ui.getDefaultSortField() != null) {
-                search.setSort(ui.getDefaultSortField());
+                search.setSorts(ui.getDefaultSortField());
```
Comment (danslaughter2) [inline db/src/main/java/com/psddev/cms/tool/search/AbstractSearchResultView.java]: Can the `Search#setSort(String)` API implementation be changed to encapsulate what `Search#setSorts(String)` is doing? At which point these 4 API calls don't need to change. Same question can be applied to `Search#getSort` vs `Search#getEffectiveSort` and then there would be 3 more unchanged API calls.

File: db/src/main/java/com/psddev/cms/tool/SearchResultRenderer.java [hunk 4]
```diff
@@ -68,23 +68,23 @@ public SearchResultRenderer(ToolPageContext page, Search search) throws IOExcept
         ToolUi ui = selectedType == null ? null : selectedType.as(ToolUi.class);
         PaginatedResult<?> result = null;
 
-        if (search.getSort() == null) {
+        if (search.getEffectiveSort() == null) {
```
Comment (danslaughter2) [inline db/src/main/java/com/psddev/cms/tool/SearchResultRenderer.java]: Same comment left on `AbstractSearchResultView.java` below.

**Comment by rohitmonga** [comment]
Closing in favor of [BSP-46](https://github.com/perfectsense/brightspot/pull/46).

**Pull request closed without merging**